packages/nx/src/native/cache/http_remote_cache.rs

```diff
@@ -0,0 +1,206 @@
+            .map_err(|e| anyhow::anyhow!(format!("Failed to finish tarball: {}", e)))?;
+        trace!("Finished tarball");
+
+        trace!("Reading tarball into memory");
+        let archive_bytes = archive.into_inner().unwrap();
+        let buffer = archive_bytes.finish()?;
+        // let mut buffer = Vec::new();
+        // // read the whole file
+        // archive_bytes.read_to_end(&mut buffer)?;
+        trace!("read tarball into memory");
+
+        let url: String = format!("{}/v1/cache/{}", self.url, hash);
+        let response = self
+            .client
+            .put(&url)
+            .header(header::CONTENT_TYPE, "application/octet-stream")
+            .body(buffer) // Convert the bytes to a Vec<u8> for the request body
+            .send()
+            .await
+            .map_err(|e| anyhow::anyhow!("Failed to send request: {}", e))?;
+
+        Ok(response.status().is_success())
+    }
+
+    async fn download_and_extract_from_result(
+        response: reqwest::Response,
+        cache_directory: String,
+    ) -> anyhow::Result<CachedResult> {
+        let content = response.bytes().await.unwrap();
+        trace!("Downloaded {} bytes from remote cache", content.len());
+        let tar = flate2::read::GzDecoder::new(content.as_ref());
+        let mut archive = Archive::new(tar);
+        let entries = archive
+            .entries() // Get the entries in the archive
+            .map_err(|_| anyhow::anyhow!("Failed to read entries from tarball"))?;
+
+        // Extract the archive to the specified cache directory
+        for entry in entries {
+            let mut entry =
+                entry.map_err(|_| anyhow::anyhow!("Failed to read entry from tarball"))?;
+            trace!(
+                "Extracting entry: {}",
+                entry
+                    .path()
+                    .map(|p| p.to_string_lossy().into_owned())
+                    .unwrap_or_else(|_| "<unknown>".to_string())
+            );
+            // Ensure the directory exists before extracting
+            if let Err(e) = entry.unpack_in(&cache_directory) {
+                return Err(anyhow::anyhow!("Failed to unpack entry: {}", e));
+            }
+        }
+
+        trace!("Extracted tarball to {}", cache_directory);
+
+        let code_path = Path::new(&cache_directory).join("code");
+        let code_file_bytes =
+            fs::read(&code_path).map_err(|_| anyhow::anyhow!("Failed to read code file"))?;
+        let code = i16::from_be_bytes([code_file_bytes[0], code_file_bytes[1]]);
+
+        trace!("Retrieved exit code from cache: {}", code);
+
+        let terminal_outputs_path = Path::new(&cache_directory).join("terminalOutput");
+        let terminal_output = fs::read_to_string(&terminal_outputs_path)
+            .map_err(|_| anyhow::anyhow!("Failed to read terminal output file"))?;
+        trace!("Retrieved terminal output from cache: {}", terminal_output);
```

Comment:
Did you take a look at storing the code and terminal output in memory so that we don't have to utilize the disk for writing and reading? It will save some cpu cycles with avoiding system calls. 

This is something we do with the self-hosted cache plugins.